### PR TITLE
Add Shufflenet support in ONNX

### PR DESCRIPTION
**Pull request by l-bat**
**Merge with extra**: https://github.com/opencv/opencv_extra/pull/523

```
force_builders=Custom
docker_image:Custom=ubuntu-openvino:16.04
buildworker:Custom=linux-2
test_opencl:Custom=ON
```

### CODE REVIEW

**Comment by dkurt** [comment]
@l-bat, please replace the base branch from `master` to `3.4`.

File: modules/dnn/test/test_onnx_importer.cpp [hunk 1]
```diff
@@ -351,6 +351,14 @@ TEST_P(Test_ONNX_nets, Inception_v1)
     testONNXModels("inception_v1", pb);
 }
 
+TEST_P(Test_ONNX_nets, Shufflenet)
+{
+    if (backend == DNN_BACKEND_INFERENCE_ENGINE &&
+         (target == DNN_TARGET_OPENCL_FP16 || target == DNN_TARGET_OPENCL))
+        throw SkipTestException("");
```
Comment (dkurt) [inline modules/dnn/test/test_onnx_importer.cpp]: This test is failed for Myriad target. Please skip it for a while.

Comment (l-bat) [reply modules/dnn/test/test_onnx_importer.cpp]: done

**Comment by dkurt** [comment]
Please consider https://github.com/opencv/opencv_extra/pull/499#pullrequestreview-158868154 in this PR.

**Comment by vpisarev** [comment]
@dkurt, can it be integrated?

File: modules/dnn/test/test_onnx_importer.cpp [hunk 2]
```diff
@@ -108,6 +108,11 @@ TEST_P(Test_ONNX_layers, BatchNormalization)
     testONNXModels("batch_norm");
 }
 
+TEST_P(Test_ONNX_layers, Transpose)
+{
+    testONNXModels("transpose");
```
Comment (dkurt) [inline modules/dnn/test/test_onnx_importer.cpp]: Tests are failed for Inference Engine backend.

**Pull request merged**